cc/position.cc

```diff
@@ -268,6 +274,7 @@ void Position::RemoveGroup(Coord c) {
 
     MG_CHECK(stones_[c].group_id() == removed_group_id);
     stones_[c] = {};
+    stone_hash_ ^= zobrist::MoveHash(c, removed_color);
```

Comment:
There's a cute trick you can do here - have each group remember the zobrist hash of each of its stones. Since zobrist hashing is associative, you can update the hash with one XOR by XORing against the hash of the entire group, instead of XOR'ing through each individual stone.

Although, now that I think about it... that also requires incrementally updating the group's hash, so it might just end up as a wash. *shrug*